test.mk

```diff
@@ -161,50 +162,256 @@ spec:
           serviceName: {{NAME}}
           servicePort: 80
 endef
+define TEST_SERVICE_SPEC_INGRESS_CLOUDEP
+$(call TEST_INGRESS)
+--- 
+$(call TEST_SERVICE)
+  targetIngress:
+    name: {{INGRESS_NAME}}
+    namespace: default
+    jwtServices:
+    - {{NAME}}
+---
+$(call TEST_APP)
+endef
+define TEST_SERVICE_SPEC_CONFIGMAP_SPEC
+apiVersion: v1
+kind: ConfigMap 
+metadata: 
+  name: swagger 
+data: 
+  spec: |
+    swagger: "2.0"
+    info:
+      description: "wildcard config for any HTTP service."
+      title: "ConfigMap Template: {{NAME}}"
+      version: "1.0.0"
+    basePath: "/"
+    consumes:
+    - "application/json"
+    produces:
+    - "application/json"
+    schemes:
+    - "http"
+    - "https"
+    paths:
+      "/**":
+        get:
+          operationId: Get
+          responses:
+            '200':
+              description: Get
+            default:
+              description: Error
+        delete:
+          operationId: Delete
+          responses:
+            '204':
+              description: Delete
+            default:
+              description: Error
+        patch:
+          operationId: Patch
+          responses:
+            '200':
+              description: Patch
+            default:
+              description: Error
+        post:
+          operationId: Post
+          responses:
+            '200':
+              description: Post
+            default:
+              description: Error
+        put:
+          operationId: Put
+          responses:
+            '200':
+              description: Put
+            default:
+              description: Error
+    {{- if .JWTAudiences }}
+    security:
+    - google_jwt: []
+    securityDefinitions:
+      google_jwt:
+        authorizationUrl: ""
+        flow: "implicit"
+        type: "oauth2"
+        x-google-issuer: "https://cloud.google.com/iap"
+        x-google-jwks_uri: "https://www.gstatic.com/iap/verify/public_key-jwk"
+        x-google-audiences: "{{ StringsJoin .JWTAudiences "," }}}"
+     {{ end }}
+    host: "{{HOSTNAME}}"
+    x-google-endpoints:
+    - name: "{{HOSTNAME}}"
+      target: "{{IP_ADDRESS}}"
+
+--- 
+$(call TEST_INGRESS)
+---
+apiVersion: ctl.isla.solutions/v1
+kind: CloudEndpoint
+metadata:
+  name: {{NAME}}
+spec:
+  project: {{PROJECT}}
+  targetIngress:
+    name: {{INGRESS_NAME}}
+    namespace: default
+    jwtServices:
+    - {{NAME}}
+  openAPISpecConfigMap: 
+    name: swagger
+    key: spec
+--- 
+$(call TEST_APP)
+endef 
+
+define CHANGED_CONFIGMAP
+apiVersion: v1
+kind: ConfigMap 
+metadata: 
+  name: swagger 
+data: 
+  spec: |
```

Comment:
Same comment  above regarding use of yaml block string. Prefer `|-` to avoid blank first line.